python/lsst/ap/pipe/ap_pipe.py

```diff
@@ -527,27 +604,73 @@ def doDiffIm(processed_repo, sciencevisit, ccdnum, templatevisit, diffim_repo):
         config.doDecorrelation = True
 
     TODO: use coadds as templates by default, not another visit (DM-11422).
+    Some of the comments in this function are placeholders for DM-11422 work.
 
     RESULT:
-    diffim_repo/deepDiff/v+sciencevisit populated with difference images
+    diffim_repo/deepDiff/v+visit populated with difference images
     and catalogs of detected sources (diaSrc, diffexp, and metadata files)
     '''
-    if os.path.exists(os.path.join(diffim_repo, 'deepDiff', 'v' + sciencevisit)):
-        print('DiffIm has already been run for visit {0}, skipping...'.format(sciencevisit))
+    lsst.log.configure()
+    log = lsst.log.Log.getLogger('ap.pipe.doDiffIm')
+    dataId_items = re.split('[ +=]', dataId)
+    dataId_dict = dict(zip(dataId_items[::2], dataId_items[1::2]))
+    if 'visit' not in dataId_dict.keys():
+        raise RuntimeError('The dataId string is missing \'visit\'')
+    else:  # save the visit number from the dataId
+        visit = dataId_dict['visit']
```

Comment:
Duplicate code with `doProcessCcd`, consider factoring.

Reply:
Point taken, but the duplication is minimal and `dataId` handling is less than ideal all around. Leaving it as-is for now.